module/applications/item/item-sheet.mjs

```diff
@@ -135,10 +135,12 @@ export default class ItemSheet5e extends ItemSheet {
     });
     context.abilityConsumptionTargets = this._getItemConsumptionTargets();
 
-    if ( "properties" in item.system ) {
-      const validProperties = CONFIG.DND5E[item.type === "weapon" ? "weaponProperties" : "physicalWeaponProperties"];
-      context.properties = Object.entries(validProperties).reduce((obj, [k, label]) => {
-        obj[k] = { label, selected: item.system.properties.has(k) };
+    if ( ("properties" in item.system) && (item.type in CONFIG.DND5E.validProperties) ) {
+      context.properties = CONFIG.DND5E.validProperties[item.type].reduce((obj, k) => {
+        obj[k] = {
+          label: CONFIG.DND5E.itemProperties[k].label,
+          selected: item.system.properties.has(k)
+        };
         return obj;
       }, {});
```

Comment:
This should resolve the sorting issue:

```suggestion
      }, {});
      if ( item.type !== "spell" ) context.properties = sortObjectEntries(context.properties, "label");
```

(just remember to import `sortObjectEntries` with `filteredKeys` at the top)